pkg/services/query/query.go

```diff
@@ -144,6 +153,7 @@ func (s *Service) handleQueryData(ctx context.Context, user *models.SignedInUser
 			if ok && idToken != "" {
 				req.Headers["X-ID-Token"] = idToken
```

Comment:
Seems like this code is now duplicated here and in the `ForwardedOAuthIdentityMiddleware`

Reply:
But if add the middleware in the next few lines doesn't it mean that all data sources will use that middleware? Like what is the point of data source even be aware of the middleware?